### DISCUSSION TITLE
cosmos as monolithic kernel

### DISCUSSION THREAD

**Discussion by AsertCreator** [CosmosOS/Cosmos#2633] [Ideas]
i came up to an idea of making cosmos a monolithic kernel stored as separate file, while os code will be stored in another file, like linux's `init` file which initializes system. this will make cosmos somewhat modular. however, this requires dynamic linker, threading and paging support. ~so, this needs to be implemented in gen3~ ok, no

P.S. some people got clr implemented in .net. but it lacks jit. so we need to compile as files to native using il2cpu (or perhaps nativeaot?) for better performance

**Comment by selkij** [comment]
nope, devs said that we should more likely just update gen2

**Comment by terminal-cs** [comment]
Gen 3 was scraped
